official/resnet/keras/keras_common.py

```diff
@@ -47,15 +55,21 @@ def __init__(self, batch_size):
     self._batch_size = batch_size
     super(TimeHistory, self).__init__()
     self.log_steps = 100
+    self.train_timestamp_history = []
 
   def on_train_begin(self, logs=None):
     self.record_batch = True
 
+  def on_train_end(self, logs=None):
+    self.train_finish_time = time.time()
+
   def on_batch_begin(self, batch, logs=None):
     if self.record_batch:
       self.start_time = time.time()
       self.record_batch = False
 
+    self.train_timestamp_history.append(BatchTimestamp(batch))
+
```

Comment:
I would do the following if I understand this correctly
- Move this under `if self.record_batch` so that index=0 will be the start time
- Also pass self.start_time rather than having a new time created, they will be almost the same but passing it will mean they are exactly the same.

Reply:
I agree on the second point: we should use the exact same timestamp. 
As to the first point, I don't think this line should be moved into the `if self.record_batch`, because if so, then we cannot warm-up (skip) for whatever number of steps we want. I think we should record the start time for *all* the batches, and end time for only a selective subset of batches (based on log_steps).